tests/platform/fw_utility/test_fwutil.py

```diff
@@ -0,0 +1,114 @@
+    command = "fwutil install chassis component {} fw -y {}".format(get_fw_path['current_component'],
+                                            os.path.join(comp_path, os.path.basename(get_fw_path['path_to_install'])))
+    try:
+        install_code = execute_update_cmd(request, dut, cmd=command, component=get_fw_path['current_component'],
+                        version_to_install=get_fw_path['version_to_install'], component_object=component_object, expected_log=FW_INSTALL_SUCCESS_LOG)
+        if install_code != SUCCESS_CODE:
+            dut.command("rm -rf {}".format(comp_path))
+            pytest.fail("Installation Failed. Aborting!")
+
+        # recover previous fw only if previous was the latest:
+        if get_fw_path['is_latest_installed'] is True:
+            dut.command("rm -rf {}".format(comp_path))
+            dut.command("mkdir -p {}".format(comp_path))
+            dut.copy(src=get_fw_path['current_fw_path'], dest=comp_path)
+            command = "fwutil install chassis component {} fw -y {}".format(get_fw_path['current_component'],
+                                                os.path.join(comp_path, os.path.basename(get_fw_path['current_fw_path'])))
+            install_code = execute_update_cmd(request, dut, cmd=command, component=get_fw_path['current_component'],
+                                version_to_install=get_fw_path['previous_ver'], component_object=component_object, expected_log=FW_INSTALL_SUCCESS_LOG)
+            if install_code != SUCCESS_CODE:
+                dut.command("rm -rf {}".format(comp_path))
+                pytest.fail("Installation Failed. Aborting!")
+
+    except Exception:
+        "Failed to install localy"
+
+    finally:
+        dut.command("rm -rf {}".format(comp_path))
+
+
+@pytest.mark.disable_loganalyzer
+def test_install_negative(request, testbed_devices, get_fw_path):
+    """
+    Tries to install invalid FW and checks the expected errors occures.
+    """
+    dut = testbed_devices['dut']
+    # invalid component name
+    cmd = 'fwutil install chassis component {} fw -y {}'.format('UNVALID_FW_NAME', get_fw_path['path_to_install'])
+    execute_wrong_command(dut, cmd, UNVALID_NAME_LOG)
+    # invalid path
+    cmd = 'fwutil install chassis component {} fw -y {}'.format(get_fw_path['current_component'], '/this/is/invalid/url')
+    execute_wrong_command(dut, cmd, UNVALID_PATH_LOG)
+    # invalid url
+    cmd = 'fwutil install chassis component {} fw -y {}'.format(get_fw_path['current_component'], 'http://not/valid/url')
+    execute_wrong_command(dut, cmd, UNVALID_URL_LOG)
+
+
+def test_update_positive(request, testbed_devices, components_list, component_object, get_fw_path, backup_platform_file):
+    """
+    Performs update from current image and from next image.
+    NOTICE: The next image should be an image with fwutil feature
+    """
+    dut = testbed_devices['dut']
+    update_from_current_img(request, dut, get_fw_path, components_list, component_object)
```

Comment:
@noaOrMlnx i think that there is no guarantee that `component_object` and `get_fw_path` results will be relevant, at least always. What if the system will randomly pick up BIOS and CPLD FW path? The variety of combinations will grow with adding a new platform components.